File: cdm/csharp/patient/conditions/SEChronicRenalStenosis.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

public class SEChronicRenalStenosis : SEPatientCondition
{
  protected SEScalar0To1 left_kidney_severity;
  protected SEScalar0To1 right_kidney_severity;

  public SEChronicRenalStenosis()
  {
    left_kidney_severity = null;
    right_kidney_severity = null;
  }

  public override void Clear()
  {
    base.Clear();
    if (left_kidney_severity != null)
      left_kidney_severity.Invalidate();
    if (right_kidney_severity != null)
      right_kidney_severity.Invalidate();
  }

  public override bool IsValid()
  {
    return HasLeftKidneySeverity() && HasRightKidneySeverity();
  }

  public bool HasLeftKidneySeverity()
  {
    return left_kidney_severity == null ? false : left_kidney_severity.IsValid();
  }
  public SEScalar0To1 GetLeftKidneySeverity()
  {
    if (left_kidney_severity == null)
      left_kidney_severity = new SEScalar0To1();
    return left_kidney_severity;
  }

  public bool HasRightKidneySeverity()
  {
    return right_kidney_severity == null ? false : right_kidney_severity.IsValid();
  }
  public SEScalar0To1 GetRightKidneySeverity()
  {
    if (right_kidney_severity == null)
      right_kidney_severity = new SEScalar0To1();
    return right_kidney_severity;
  }

}

File: src/java/com/kitware/pulse/cdm/patient/conditions/SEChronicRenalStenosis.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.patient.conditions;

import com.kitware.pulse.cdm.bind.PatientConditions.ChronicRenalStenosisData;
import com.kitware.pulse.cdm.properties.SEScalar0To1;

public class SEChronicRenalStenosis extends SEPatientCondition
{

  private static final long serialVersionUID = -4917822281366252892L;
  protected SEScalar0To1 leftKidneySeverity;
  protected SEScalar0To1 rightKidneySeverity;
  
  public SEChronicRenalStenosis()
  {
    leftKidneySeverity = null;
    rightKidneySeverity = null;
  }
  
  @Override
  public void reset()
  {
    super.reset();
    if (leftKidneySeverity != null)
      leftKidneySeverity.invalidate();
    if (rightKidneySeverity != null)
      rightKidneySeverity.invalidate();
  }
  
  public void copy(SEChronicRenalStenosis other)
  {
    if(this==other)
      return;
    super.copy(other);
    if (other.leftKidneySeverity != null)
      getLeftKidneySeverity().set(other.getLeftKidneySeverity());
    else if (leftKidneySeverity != null)
      leftKidneySeverity.invalidate();
    
    if (other.rightKidneySeverity != null)
      getRightKidneySeverity().set(other.getRightKidneySeverity());
    else if (rightKidneySeverity != null)
      rightKidneySeverity.invalidate();
  }
  
  public static void load(ChronicRenalStenosisData src, SEChronicRenalStenosis dst) 
  {
    SEPatientCondition.load(src.getPatientCondition(), dst);
    if(src.hasLeftKidneySeverity())
      SEScalar0To1.load(src.getLeftKidneySeverity(),dst.getLeftKidneySeverity());
    if(src.hasRightKidneySeverity())
      SEScalar0To1.load(src.getRightKidneySeverity(),dst.getRightKidneySeverity());
  }
  
  public static ChronicRenalStenosisData unload(SEChronicRenalStenosis src) 
  {
    ChronicRenalStenosisData.Builder dst = ChronicRenalStenosisData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  
  protected static void unload(SEChronicRenalStenosis src, ChronicRenalStenosisData.Builder dst)
  {
    SEPatientCondition.unload(src, dst.getPatientConditionBuilder());
    if (src.hasLeftKidneySeverity())
      dst.setLeftKidneySeverity(SEScalar0To1.unload(src.leftKidneySeverity));
    if (src.hasRightKidneySeverity())
      dst.setRightKidneySeverity(SEScalar0To1.unload(src.rightKidneySeverity));
  }
  
  public boolean hasLeftKidneySeverity()
  {
    return leftKidneySeverity == null ? false : leftKidneySeverity.isValid();
  }
  public SEScalar0To1 getLeftKidneySeverity()
  {
    if (leftKidneySeverity == null)
      leftKidneySeverity = new SEScalar0To1();
    return leftKidneySeverity;
  }
  
  public boolean hasRightKidneySeverity()
  {
    return rightKidneySeverity == null ? false : rightKidneySeverity.isValid();
  }
  public SEScalar0To1 getRightKidneySeverity()
  {
    if (rightKidneySeverity == null)
      rightKidneySeverity = new SEScalar0To1();
    return rightKidneySeverity;
  }
  
  @Override
  public String toString()
  {
    return "Chronic Renal Stenosis" 
        + "\n\tLeftKidneySeverity: " + getLeftKidneySeverity()
        + "\n\tRightKidneySeverity: " + getRightKidneySeverity();
  }

  
}
